test/integration/apiserver/apiserver_test.go

```diff
@@ -3268,6 +3273,279 @@ func TestDisableEmulationVersion(t *testing.T) {
+				"ValidatingAdmissionPolicy":                      1,
+				"VolumeAttributesClass":                          0,
+				"VolumeCapacityPriority":                         0,
+				"WatchBookmark":                                  1,
+				"WatchCacheInitializationPostStartHook":          0,
+				"WatchFromStorageWithoutResourceVersion":         0,
+				"WatchList":                                      0,
+				"WatchListClient":                                0,
+				"WinDSR":                                         0,
+				"WinOverlay":                                     1,
+				"WindowsHostNetwork":                             1,
+				"ZeroLimitedNominalConcurrencyShares":            1,
+			},
+			previousFeatureGatesWithLockToDefaultTrue: map[string]struct{}{
+				"AppArmor":                                {},
+				"AppArmorFields":                          {},
+				"CloudDualStackNodeIPs":                   {},
+				"CPUManager":                              {},
+				"DisableCloudProviders":                   {},
+				"DisableKubeletCloudCredentialProviders":  {},
+				"DevicePluginCDIDevices":                  {},
+				"HPAContainerMetrics":                     {},
+				"JobPodFailurePolicy":                     {},
+				"KubeProxyDrainingTerminatingNodes":       {},
+				"LegacyServiceAccountTokenCleanUp":        {},
+				"LogarithmicScaleDown":                    {},
+				"MinDomainsInPodTopologySpread":           {},
+				"NewVolumeManagerReconstruction":          {},
+				"NodeOutOfServiceVolumeDetach":            {},
+				"PDBUnhealthyPodEvictionPolicy":           {},
+				"PersistentVolumeLastPhaseTransitionTime": {},
+				"PodDisruptionConditions":                 {},
+				"PodHostIPs":                              {},
+				"PodSchedulingReadiness":                  {},
+				"ElasticIndexedJob":                       {},
+				"StableLoadBalancerNodeSet":               {},
+				"StatefulSetStartOrdinal":                 {},
+				"AdmissionWebhookMatchConditions":         {},
+				"AggregatedDiscoveryEndpoint":             {},
+				"APIListChunking":                         {},
+				"EfficientWatchResumption":                {},
+				"KMSv2":                                   {},
+				"KMSv2KDF":                                {},
+				"RemainingItemCount":                      {},
+				"ServerSideApply":                         {},
+				"ServerSideFieldValidation":               {},
+				"WatchBookmark":                           {},
+				"ZeroLimitedNominalConcurrencyShares":     {},
+				"ValidatingAdmissionPolicy":               {},
+			},
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.emulationVersion, func(t *testing.T) {
+			featuregatetesting.SetFeatureGateEmulationVersionDuringTest(t, utilfeature.DefaultFeatureGate, version.MustParse(tc.emulationVersion))
+			server := kubeapiservertesting.StartTestServerOrDie(t,
+				&kubeapiservertesting.TestServerInstanceOptions{BinaryVersion: "1.32"},
```

Comment:
Is BinaryVersion necessary? When HEAD hits 1.33, are we trying to remain on 1.32 in this test?

Reply:
I based these values off of another `--emulated-version` tests (which this one is modeled after) which used a hardcoded 1.32 (which was n+1, now it is n) and compared it to a prev version (which is 1.31 for these tests).  For example an already existing `--emulated-version` test does an identical setup for these values to what this test does w.r.t. what version it sets as default and what is passes in as a prev version (1.31) - https://github.com/kubernetes/kubernetes/blob/master/test/integration/apiserver/apiserver_test.go#L3149-L3208

I have removed the 1.32 value for now so that it uses the default which should be the correct binary version of gates for the branch